facia-purger/src/main/scala/com/gu/purge/facia/ParseS3Path.scala

```diff
@@ -0,0 +1,20 @@
+package com.gu.purge.facia
+
+import org.parboiled2.{ Parser, ParserInput, Rule1 }
+
+import scala.util.Success
+
+// /aws-frontend-store/CODE/frontsapi/pressed/live/au/fapi/pressed.json
+class ParseS3Path(stage: String, val input: ParserInput) extends Parser {
+  def prefix = rule { s"$stage/frontsapi/pressed/live/" }
+  val suffix = "/fapi/pressed.json"
+
+  // Capture the front by using a negative lookahead for the suffix
+  def front = rule { capture(oneOrMore(!suffix ~ ANY)) }
+  def expr = rule { prefix ~ front ~ suffix ~ EOI }
+
+  def apply() = expr.run() match {
+    case Success(matched) => Some(matched)
+    case x => { println(s"error: $x"); None }
```

Comment:
yeah I think B would be nice, since the log will generally be short we probably don't need a powerful system with many parameters stored.
Cloudformation is another missing thing that we'll need to make this prod ready, otherwise there's no way to manage the lambda if the stack changes.